src/main/java/emissary/place/ServiceProviderPlace.java

```diff
@@ -925,6 +928,72 @@ protected void deregisterFromDirectory(List<String> keys) {
         }
     }
 
+    /**
+     * Reset the keys and reload the {@link Configurator}
+     *
+     * @throws IOException if there is an issue loading the config
+     */
+    protected void refresh() throws IOException {
```

Comment:
It occurs to me that this might not be safe to call while the place is actively processing an IBDO, or while trying to identify the next place that should process a payload.